catalog/spatial/registry/registry-admin-modules/registry-admin-local-ui/src/main/webapp/js/view/Registry.view.js

```diff
@@ -272,21 +283,92 @@ define([
                     data = this.model.toJSON();
                 }
                 data.name = this.getNodeName();
+                data.remote = this.isRemote();
                 return data;
            },
            getNodeName: function() {
                return this.model.get('name');
+           },
+           isRemote: function() {
+                var nodeName = this.getNodeName();
+                var remoteNames = this.model.collection.getRemoteNodes().map(
+                    function (n) {
+                        return n.get('name');
+                    }
+                );
+                // return true if the node name is in the remote names
+                return ($.inArray(nodeName, remoteNames) >= 0);
            }
         });
 
+        RegistryView.DeleteAllAdditionalModal = RegistryView.DeleteModal.extend({
+            deleteNode: function() {
+                var registryIds = this.model.getSecondaryNodes().map(
+                    function (n) {
+                        return n.get('registryId');
+                    }
+                );
+                this.model.deleteNodes(registryIds);
+                this.close();
+            },
+            serializeData: function() {
+                var data = {};
+                if (this.model) {
+                    data = this.model.toJSON();
+                }
+                if (this.model.getSecondaryNodes().length === 0) {
+                    data.noNodes = true;
+                }
+                data.deleteAll = true;
+                data.name = "Additional Local";
+
+                return data;
+            }
+        });
+
+        RegistryView.DeleteAllRemoteModal = RegistryView.DeleteModal.extend({
```

Comment:
I think you could just have one DeleteModal that takes the nodes and title as options. This would reduce duplicate code and make things a little simpler.